Source/WebCore/css/CSSVariableReferenceValue.cpp

```diff
@@ -139,16 +158,19 @@ bool CSSVariableReferenceValue::resolveTokenRange(CSSParserTokenRange range, Vec
         }
         tokens.append(range.consume());
     }
-    return success;
+    if (!success)
+        return { };
+
+    return tokens;
 }
 
 RefPtr<CSSVariableData> CSSVariableReferenceValue::resolveVariableReferences(Style::BuilderState& builderState) const 
 {
-    Vector<CSSParserToken> resolvedTokens;
-    if (!resolveTokenRange(m_data->tokenRange(), resolvedTokens, builderState))
+    auto resolvedTokens = resolveTokenRange(m_data->tokenRange(), builderState);
+    if (!resolvedTokens)
         return nullptr;
 
-    return CSSVariableData::create(resolvedTokens);
+    return CSSVariableData::create(*resolvedTokens);
```

Comment:
Seems unfortunate we have to copy here. Can we add a CSSVariableData::create() overload that takes an r-value and move the resolved tokens in? Or does the processing done in CSSVariableData's constructor make this a meaning less optimization?

Reply:
Maybe something to do separately. I did notice it and wondered about the same things you did.